cmd/lotus-shed/chain_index.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+	lcli "github.com/filecoin-project/lotus/cli"
+)
+
+func withCategory(cat string, cmd *cli.Command) *cli.Command {
+	cmd.Category = strings.ToUpper(cat)
+	return cmd
+}
+
+var chainIndexCmds = &cli.Command{
+	Name:            "chainindex",
+	Usage:           "Commands related to managing the chainindex",
+	HideHelpCommand: true,
+	Subcommands: []*cli.Command{
+		withCategory("chainindex", validateBackfillChainIndexCmd),
+	},
+}
+
+var validateBackfillChainIndexCmd = &cli.Command{
+	Name:  "validate-backfill",
+	Usage: "Validates and optionally backfills the chainindex for a range of epochs",
+	Description: `
+lotus-shed chainindex validate-backfill --from <start_epoch> --to <end_epoch> [--backfill] [--log-good]
+
+The command validates the chain index entries for each epoch in the specified range, checking for missing or
+inconsistent entries (i.e. the indexed data does not match the actual chain state). If '--backfill' is enabled
+(which it is by default), it will attempt to backfill any missing entries using the 'ChainValidateIndex' API.
+
+Parameters:
+  - '--from' (required): The starting epoch (inclusive) for the validation range. Must be greater than 0.
```

Comment:
Hey, just reading the docs and thinking through this, we should make this optional and have it default to the current head if it's not supplied (like current indexes.go functionality). The most common case for this is going to be to catch the node up, and you're almost always going to be wanting to catch up to the head. So to avoid figuring out what the head is, let's figure it out for them.

Reply:
@rvagg I worry about them messing up book-keeping because they don't know what epoch head is at right now. Making this explicit atleast makes them think about the range they're backfilling.

Happy to change it if you have strong opinions on this -> otherwise -> let's do it if a user asks for it.